### DISCUSSION TITLE
How to make a select2 readonly?

### DISCUSSION THREAD

**Discussion by realtebo** [Laravel-Backpack/community-forum#649] [Q&A (Help)] [answered]
I have a select2 with these attributes

'attributes' => [
                    'disabled' => 'disabled',
                    'readonly' => 'readonly'
                ],

disabled is working, but I cannot use this because browser will not send its value as payload of the POST

but readonly is not working. user can still change selected value !

**Comment by pxpm** [comment, marked as answer]
Hey @realtebo there are some issues with readonly that we are aware. We started adding them to the crud api here: [#4443](https://github.com/Laravel-Backpack/CRUD/pull/4443) but noticed that alot of fields required additional customizations for the read only, especially the PRO ones.

1 - with readonly the user can still navigate with the keyboard unless we implement keyboard blockers too. 
2 - some fields couldn't be "readonly" and we needed to disable them anyways to simulate readonly. 

This is the screenshoot from PRO fields like select2 when we attempt to add the readonly: 
![image](https://github.com/Laravel-Backpack/community-forum/assets/7188159/0bfd0a20-09de-447e-995c-0d0b86796fb5)

We decided to let the idea rest for a while. 

So what I would advise you for now, if you really need to keep the field a select2 readonly is to either:
- add that bit of js in a custom field `php artisan backpack:field select2_readonly --from=select2` 
- do the same using the CRUD JS API, by registering that event listener in the field. 

```javascript
crud.field('select2').onChange(function(field) {
 // field.$input is the jquery wrapped element
field.$input.on( ... );
}).change();
```

We maybe revisit the reaonly PR's in future, in that case you can later delete the custom file or the crud js script and use the core functionality for it.

Cheers

**Comment by realtebo** [comment]
The problem is raised here due to another bug about form operation and morph fields 

I open a new issue in a few minutes. 

Actually we need a workaround because it's vital for our project [long sad story as always due to stupidity of customers]
